keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -0,0 +1,1073 @@
+  policy.
+- A privileged pod should never be allowed by baseline or restricted
+- A Fully specified pod within the bounds of baseline should be allowed by baseline, and rejected by
+  restricted.
+- A minimally specified restricted pod should be allowed at a pinned version.
+
+**Integration Tests:** Audit mode tests should be added to integration testing, where we have
+existing audit logging tests.
+
+**Manual Testing Resources:** Pod resources will be provided covering all dimensions of the baseline
+& restricted profiles, for validation of 3rd party policy implementations. These have been drafted
+by @JimBugwadia: https://github.com/JimBugwadia/pod-security-tests
+
+**Unit Tests:** Both the library and admission controller implementations will have thorough
+coverage of unit tests.
+
+### Monitoring
+
+A single metric will be added to track policy evaluations against pods and [templated pods].
+[Namespace evaluations](#namespace-policy-update-warnings) are not counted.
+
+```
+<component_name>_evaluations_total
+```
+
+The metric will use the following labels:
+
+1. `decision {allow, deny, exempt, error}` - The policy decision. Error is reserved for panics or
+   other errors in policy evaluation. Update requests that are out of scope (see [Updates](#updates)
+   above) are not counted.
+3. `policy_level {privileged, baseline, restricted}` - The policy level that the request was
+   evaluated against.
+4. `policy_version {latest, v1.YY, >v1.ZZ}` - The policy version that was used for the evaluation.
+   How to constrain cardinality is unresolved (see below).
+5. `mode {enforce, warn, audit}` - The type of evaluation mode being recorded. Note that a single
+   request can increment this metric 3 times, once for each mode. If this admission controller is
+   enabled, every every create request and in-scope update request will at least increment the
+   `enforce` total.
+6. `request_operation {create, update}` - The operation of the request being checked.
+
+<<[UNRESOLVED]>>
+
+_Non-blocking: can be decided on the implementing PR_
+
+How should policy version labels be handled, to control cardinality? Specifically:
+- How should future versions be labeled?
+- How should (very old) past versions be labeled?
+
+Ideas:
+- If the version is set higher than `v{latest+1}`, then `>v{latest+1}`
+  will be used. In other words, if the current version of the admission controller is v1.22, then a
+  version of `v1.23` would be unchanged, but `v1.24` would be recorded as `>v1.23`.
+    - Concern that the sliding-window approach will cause issues with historical data.
+- If the version is set higher than latest, simply record it as `future`. Allow recording of all
+  past versions.
+
+<<[/UNRESOLVED]>>
+
+### Audit Annotations
+
+The following audit annotations will be added:
+
+1. `pod-security.kubernetes.io/enforce-policy = <policy_level>:<resolved_version>` Record which policy was evaluated
```

Comment:
It cuts down on the size & verbosity of the logs. It's useful to keep them separate on labels because of the limitations of label selectors, but most log processors that I've seen can handle separating these values if need be. Is there a reason you'd want to see them separated?

Reply:
This just seemed inconsistent with the labels used on namespaces.